src/servers/http_server.cc

```diff
@@ -624,791 +458,140 @@ HTTPAPIServer::HandleStatus(evhtp_request_t* req, const std::string& status_uri)
-
-    // If 'byte_size' is zero then need to add an empty input data
-    // block... the provider expects at least one data block for every
-    // input.
-    if (byte_size == 0) {
-      RETURN_IF_ERR(TRTSERVER_InferenceRequestProviderSetInputData(
-          request_provider, io.name().c_str(), nullptr, 0 /* byte_size */,
-          TRTSERVER_MEMORY_CPU, 0 /* memory_type_id */));
-    } else {
-      // If input is in shared memory then verify that the size is
-      // correct and set input from the shared memory.
-      if (io.has_shared_memory()) {
-        if (byte_size != io.shared_memory().byte_size()) {
-          return TRTSERVER_ErrorNew(
-              TRTSERVER_ERROR_INVALID_ARG,
-              std::string(
-                  "unexpected shared-memory size " +
-                  std::to_string(io.shared_memory().byte_size()) +
-                  " for input '" + io.name() + "', expecting " +
-                  std::to_string(byte_size) + " for model '" + model_name + "'")
-                  .c_str());
-        }
-
-        void* base;
-        TRTSERVER_Memory_Type memory_type = TRTSERVER_MEMORY_CPU;
-        int64_t memory_type_id;
-        TRTSERVER_SharedMemoryBlock* smb = nullptr;
-        RETURN_IF_ERR(smb_manager_->Get(&smb, io.shared_memory().name()));
-        RETURN_IF_ERR(TRTSERVER_ServerSharedMemoryAddress(
-            server_.get(), smb, io.shared_memory().offset(),
-            io.shared_memory().byte_size(), &base));
-        TRTSERVER_SharedMemoryBlockMemoryType(smb, &memory_type);
-        TRTSERVER_SharedMemoryBlockMemoryTypeId(smb, &memory_type_id);
-        RETURN_IF_ERR(TRTSERVER_InferenceRequestProviderSetInputData(
-            request_provider, io.name().c_str(), base, byte_size, memory_type,
-            memory_type_id));
-      } else {
-        while ((byte_size > 0) && (v_idx < n)) {
-          char* base = static_cast<char*>(v[v_idx].iov_base);
-          size_t base_size;
-          if (v[v_idx].iov_len > byte_size) {
-            base_size = byte_size;
-            v[v_idx].iov_base = static_cast<void*>(base + byte_size);
-            v[v_idx].iov_len -= byte_size;
-            byte_size = 0;
-          } else {
-            base_size = v[v_idx].iov_len;
-            byte_size -= v[v_idx].iov_len;
-            v_idx++;
-          }
-
-          RETURN_IF_ERR(TRTSERVER_InferenceRequestProviderSetInputData(
-              request_provider, io.name().c_str(), base, base_size,
-              TRTSERVER_MEMORY_CPU, 0 /* memory_type_id */));
-        }
-
-        if (byte_size != 0) {
-          return TRTSERVER_ErrorNew(
-              TRTSERVER_ERROR_INVALID_ARG,
-              std::string(
-                  "unexpected size for input '" + io.name() + "', expecting " +
-                  std::to_string(byte_size) + " bytes for model '" +
-                  model_name + "'")
-                  .c_str());
-        }
-      }
-    }
-  }
+  if (!exit_loop) {
```

Comment:
Don't you need a separate thread to run the event loop? Otherwise won't you be looping here until the server is stopped?

Reply:
Yes. I will move this to a separate thread.